src/NHibernate/Util/ADOExceptionReporter.cs

```diff
@@ -19,20 +19,25 @@ public static void LogExceptions(Exception ex)
 
 		public static void LogExceptions(Exception ex, string message)
 		{
-			if (log.IsErrorEnabled)
+			if (log.IsErrorEnabled())
 			{
-				if (log.IsDebugEnabled)
+				if (log.IsDebugEnabled())
 				{
 					message = StringHelper.IsNotEmpty(message) ? message : DefaultExceptionMsg;
-					log.Debug(message, ex);
+					log.Debug(ex, message);
 				}
+
+				// Pass full exception on highest call
+				if (log.IsWarnEnabled()) log.Warn(ex, ex.ToString());
+				log.Error(ex, ex.Message);
+				ex = ex.InnerException;
 				while (ex != null)
 				{
-					log.Warn(ex);
+					if (log.IsWarnEnabled()) log.Warn(ex.ToString());
```

Comment:
I would instead change that to `log.Warn(ex, ex.Message)` and revert the other changes.